core/object/script_language.cpp

```diff
@@ -268,12 +268,23 @@ void ScriptServer::global_classes_clear() {
 
 void ScriptServer::add_global_class(const StringName &p_class, const StringName &p_base, const StringName &p_language, const String &p_path) {
 	ERR_FAIL_COND_MSG(p_class == p_base || (global_classes.has(p_base) && get_global_class_native_base(p_base) == p_class), "Cyclic inheritance in script class.");
-	GlobalScriptClass g;
-	g.language = p_language;
-	g.path = p_path;
-	g.base = p_base;
-	global_classes[p_class] = g;
-	inheriters_cache_dirty = true;
+	if (GlobalScriptClass *existing = global_classes.getptr(p_class)) {
```

Comment:
```suggestion
	GlobalScriptClass *existing = global_classes.getptr(p_class);
	if (existing) {
```
We usually don't use this syntax as it's less readable